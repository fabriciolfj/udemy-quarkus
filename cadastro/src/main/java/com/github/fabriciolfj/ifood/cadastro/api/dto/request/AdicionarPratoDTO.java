package com.github.fabriciolfj.ifood.cadastro.api.dto.request;

public class AdicionarPratoDTO {

    public String nome;
    public String descricao;
}
