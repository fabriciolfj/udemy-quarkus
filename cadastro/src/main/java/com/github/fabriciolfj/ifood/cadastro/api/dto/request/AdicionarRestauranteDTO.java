package com.github.fabriciolfj.ifood.cadastro.api.dto.request;

public class AdicionarRestauranteDTO {

    public String proprietario;
    public String nome;
    public LocalizacaoDTO localizacao;
}
