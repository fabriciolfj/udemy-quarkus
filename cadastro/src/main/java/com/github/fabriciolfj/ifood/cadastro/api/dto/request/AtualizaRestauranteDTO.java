package com.github.fabriciolfj.ifood.cadastro.api.dto.request;

public class AtualizaRestauranteDTO {

    public String nome;
}
