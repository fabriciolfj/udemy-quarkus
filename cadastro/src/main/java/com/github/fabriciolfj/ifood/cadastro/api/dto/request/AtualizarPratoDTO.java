package com.github.fabriciolfj.ifood.cadastro.api.dto.request;

import java.math.BigDecimal;

public class AtualizarPratoDTO {

    public BigDecimal preco;
}
