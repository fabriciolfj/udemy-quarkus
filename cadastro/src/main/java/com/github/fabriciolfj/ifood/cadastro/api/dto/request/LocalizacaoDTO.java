package com.github.fabriciolfj.ifood.cadastro.api.dto.request;

public class LocalizacaoDTO {

    public Double latitude;
    public Double longitude;
}
