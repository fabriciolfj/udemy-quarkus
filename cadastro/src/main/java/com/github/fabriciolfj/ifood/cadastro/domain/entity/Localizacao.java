package com.github.fabriciolfj.ifood.cadastro.domain.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "localizacao")
public class Localizacao extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public Double latitude;

    public Double longitude;
}
