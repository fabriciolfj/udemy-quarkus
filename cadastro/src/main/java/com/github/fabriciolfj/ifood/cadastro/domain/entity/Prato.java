package com.github.fabriciolfj.ifood.cadastro.domain.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.math.BigDecimal;

@Entity
@Table(name = "prato")
public class Prato extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String nome;

    public String descricao;

    @ManyToOne
    public Restaurante restaurante;

    public BigDecimal preco;
}
