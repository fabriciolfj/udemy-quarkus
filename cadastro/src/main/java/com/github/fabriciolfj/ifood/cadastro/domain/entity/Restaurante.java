package com.github.fabriciolfj.ifood.cadastro.domain.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.util.Date;

@Entity
@Table(name = "restaurante")
public class Restaurante extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String proprietario;

    public String cnpj;

    public String nome;

    @OneToOne(cascade = CascadeType.ALL)
    public Localizacao localizacao;

    @CreationTimestamp
    public Date dataCriacao;

    @UpdateTimestamp
    public Date dataAtualizacao;

}
