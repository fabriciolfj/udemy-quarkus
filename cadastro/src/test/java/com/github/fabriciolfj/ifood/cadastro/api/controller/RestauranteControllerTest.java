package com.github.fabriciolfj.ifood.cadastro.api.controller;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.fabriciolfj.ifood.cadastro.CadastroTestLifecycleManager;
import com.github.fabriciolfj.ifood.cadastro.api.dto.request.AdicionarRestauranteDTO;
import com.github.fabriciolfj.ifood.cadastro.domain.entity.Restaurante;
import com.github.fabriciolfj.ifood.cadastro.util.TokenUtils;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.Header;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteControllerTest {

    private String token;

    @BeforeEach
    public void setup() throws Exception {
        token = TokenUtils.generateTokenString("/JWTProprietarioClaims.json", null);
    }

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testBuscarRestaurantes() {
        final String resultado = RestAssured.given()
                .when()
                .header(new Header("Authorization", "Bearer " + token))
                .get("/restaurantes")
                .then()
                .statusCode(200)
               .extract().asString();

        Approvals.verifyJson(resultado);
    }

    @Test
    public void testAdicionarRestaurante() {
        var restaurante = new AdicionarRestauranteDTO();
        restaurante.nome = "Restaurante 3";
        restaurante.proprietario = "Fabricio";
        restaurante.cnpj = "68.562.575/0001-31";

        RestAssured.given()
                .contentType("application/json")
                .header(new Header("Authorization", "Bearer " + token))   
                .body(restaurante)
                .post("/restaurantes")
                .then()
                .statusCode(201);
    }

    /*
    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testAlterarRestaurante() {
        AtualizarRestauranteDTO dto = new AtualizarRestauranteDTO();
        dto.nomeFantasia = "novoNome";
        Long parameterValue = 123L;
        given()
                .with().pathParam("id", parameterValue)
                .body(dto)
                .when().put("/restaurantes/{id}")
                .then()
                .statusCode(Status.NO_CONTENT.getStatusCode())
                .extract().asString();

        Restaurante findById = Restaurante.findById(parameterValue);

        //poderia testar todos os outros atribudos
        Assert.assertEquals(dto.nomeFantasia, findById.nome);
    }*/

}
