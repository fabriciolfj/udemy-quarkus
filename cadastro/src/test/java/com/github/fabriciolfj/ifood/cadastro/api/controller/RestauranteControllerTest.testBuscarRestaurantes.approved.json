[
  {
    "id": 123,
    "nome": "Novo restaurante",
    "proprietario": "Id do keyclock"
  },
  {
    "id": 234,
    "nome": "Novo Restaurante 2",
    "proprietario": "ID do keyclock2"
  }
]